test/unittest/tests/helpers/vector_bool.cpp

```diff
@@ -0,0 +1,139 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *      All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "vector_bool.h"
+#include <algorithm>
+#include <charconv>
+#include <iostream>
+
+std::vector<bool>
+vector_bool_from_hex_string(std::string const &hex_str)
+{
+    std::vector<bool> result;
+
+    for (auto iter = hex_str.crbegin(); iter != hex_str.crend(); ++iter) {
+        int value_of_hex_char = std::stoi(std::string(1, *iter), nullptr, 16);
+        result.push_back((value_of_hex_char & 1) != 0);
+        result.push_back((value_of_hex_char & 2) != 0);
+        result.push_back((value_of_hex_char & 4) != 0);
+        result.push_back((value_of_hex_char & 8) != 0);
+    }
+
+    return result;
+}
+
+std::string
+vector_bool_to_hex_string(std::vector<bool> const &vector_bool)
+{
+    std::string result;
+
+    auto iter = vector_bool.cbegin();
+    while (iter != vector_bool.cend()) {
+        int bit_count = 0;
+        int hex_digit_value = 0;
+        while ((bit_count < 4) && (iter != vector_bool.cend())) {
+            int bit = *iter++;
+            hex_digit_value |= (bit & 1) << bit_count;
+            bit_count++;
+        }
+        char hex_digit = 0;
+        std::to_chars(&hex_digit, &hex_digit + 1, hex_digit_value, 16);
+        result.push_back(hex_digit);
+    }
+
+    std::reverse(result.begin(), result.end());
+    return result;
+}
+
+std::vector<bool>
+vector_bool_from_binary_string(std::string const &binary_str)
+{
+    std::vector<bool> result;
+
+    for (auto iter = binary_str.crbegin(); iter != binary_str.crend(); iter++) {
+        char ch = *iter;
+        bool bit = false;
+        switch (ch) {
+        case '0':
+            bit = false;
+            break;
+        case '1':
+            bit = true;
+            break;
+        default:
+            throw(std::invalid_argument("Binary value is not 0 or 1"));
+        }
+        result.push_back(bit);
+    }
+
+    return result;
+}
+
+std::string
+vector_bool_to_binary_string(std::vector<bool> const &vector_bool)
+{
+    std::string result;
+
+    for (auto iter = vector_bool.crbegin(); iter != vector_bool.crend(); iter++) {
+        char ch = *iter ? '1' : '0';
+        result.push_back(ch);
+    }
+
+    return result;
+}
+
+std::vector<bool>
+operator&(std::vector<bool> const &a, std::vector<bool> const &b)
```

Comment:
This operation semantic is similar but different to that of integral values, for example:
```
0x1 & 0xFF == 0x1 (integer)
0x1 & 0xFF == 0xFF (vector<bool>)
```
Which may be surprising. I think this needs to be considered. A possibility is to make the operator invalid for  `a.size() != b.size()`.

Reply:
I've added a test case that shows that 0x01 & 0xff == 0x01 as expected. 

```
        std::vector<bool> v1(vector_bool_from_hex_string("1"));
        std::vector<bool> v2(vector_bool_from_hex_string("ff"));

        REQUIRE(get_true_count(v1) == 1);
        REQUIRE(get_true_count(v2) == 8);

        std::vector<bool> v_and(v1 & v2);
        std::vector<bool> v_xor(v1 ^ v2);

        REQUIRE(v_and == vector_bool_from_hex_string("01"));
        REQUIRE(v_xor == vector_bool_from_hex_string("fe"));
```

Reply:
@jeremythorp Thanks for that, I'm still curious about the following behavior:
```
    Section("Behavior Question")
    {
        std::vector<bool> v1(vector_bool_from_hex_string("1"));
        std::vector<bool> v2;
        std::vector<bool> v3;

        REQUIRE(get_true_count(v1) == 1);
        REQUIRE(get_true_count(v2) == 0);
        REQUIRE(get_true_count(v3) == 0);

        std::vector<bool> v1_and_v2(v1 & v2);
        std::vector<bool> v2_and_v3(v2 & v3);

        REQUIRE(get_true_count(v2_and_v3) == 0);
        REQUIRE(v2_and_v3.size() == 0);
        REQUIRE(get_true_count(v1_and_v2) == 0);
        REQUIRE(v1_and_v2.size() == 0);
    }
```
The last test `REQUIRE(v1_and_v2.size() == 0);` fails. It's not a logical bug, but the behavior is different.